### PR TITLE
Templatize resourceLabels for standardization

### PR DESCRIPTION
**Pull request by sahilvv**
## What was changed
This PR addresses Issue 519 - https://github.com/temporalio/helm-charts/issues/519

## Why?
To enable consistency of labels across the temporal helm chart codebase and enable the right selectors, this PR aims to use helper functions to achieve the same. 

## Checklist
1. Closes (https://github.com/temporalio/helm-charts/issues/519)

2. How was this tested:
```
helm install --dry-run temporal .
helm install temporal .
```

3. Any docs updates needed?
N/A

### CODE REVIEW

File: charts/temporal/templates/admintools-deployment.yaml [hunk 1]
```diff
@@ -21,16 +16,8 @@ spec:
...
       labels:
-        app.kubernetes.io/name: {{ include "temporal.name" . }}
-        helm.sh/chart: {{ include "temporal.chart" . }}
-        app.kubernetes.io/managed-by: {{ .Release.Service }}
-        app.kubernetes.io/instance: {{ .Release.Name }}
-        app.kubernetes.io/version: {{ .Chart.AppVersion | replace "+" "_" }}
+        {{ include "temporal.resourceLabels" . | nindent 8 }}
         app.kubernetes.io/component: admintools
-        app.kubernetes.io/part-of: {{ .Chart.Name }}
-        {{- with $.Values.admintools.podLabels }}
```
Comment (robholland) [inline charts/temporal/templates/admintools-deployment.yaml:31]: This functionality needs to stay. I suggest passing scope and resource type arguments to the helper, which will then check `.Values.${scope}.additionalLabels` and `.Values.${scope}.${type}Labels` alongside the current `.Values.addtionalLabels`.

**Review by robholland — CHANGES_REQUESTED** [review]

File: charts/temporal/templates/admintools-deployment.yaml (hunk 1 above)
Comment (sahilvv) [reply charts/temporal/templates/admintools-deployment.yaml:31]: addressed @robholland . PTAL.

File: charts/temporal/templates/_helpers.tpl [hunk 2]
```diff
@@ -58,6 +58,61 @@ and we want to make sure that the component is included in the name.
...
+app.kubernetes.io/name: {{ include "temporal.name" $global }}
+helm.sh/chart: {{ include "temporal.chart" $global }}
+app.kubernetes.io/managed-by: {{ index $global "Release" "Service" }}
+app.kubernetes.io/instance: {{ index $global "Release" "Name" }}
+app.kubernetes.io/version: {{ include "temporal.appVersion" $global }}
+app.kubernetes.io/part-of: {{ $global.Chart.Name }}
+{{- with $resourceType -}}
+{{- $resourceTypeKey := printf "%sLabels" . -}}
+{{- $resourceLabels := dict -}}
+{{ if or (eq $scope "") (ne $component "server") -}}
```
Comment (robholland) [inline charts/temporal/templates/_helpers.tpl:95]:
```suggestion
{{- if or (eq $scope "") (ne $component "server") -}}
```

File: charts/temporal/templates/_helpers.tpl [hunk 3]
```diff
@@ -58,6 +58,61 @@ and we want to make sure that the component is included in the name.
...
+{{- $global := index . 0 -}}
+{{- $scope := index . 1 -}}
+{{- $resourceType := index . 2 -}}
+{{- $component := "server" -}}
+{{- if (or (eq $scope "admintools") (eq $scope "web")) -}}
+{{- $component = $scope -}}
+{{- end -}}
+{{- with $scope -}}
+app.kubernetes.io/component: {{ . }}
+{{- end }}
```
Comment (robholland) [inline charts/temporal/templates/_helpers.tpl:85]:
```suggestion
{{ end -}}
```

This avoids a blank line in the manifest if we don't print the component.

File: charts/temporal/templates/admintools-deployment.yaml [hunk 4]
```diff
@@ -4,13 +4,7 @@ kind: Deployment
...
   name: {{ include "temporal.componentname" (list . "admintools") }}
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" . }}
-    helm.sh/chart: {{ include "temporal.chart" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/version: {{ .Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/component: admintools
-    app.kubernetes.io/part-of: {{ .Chart.Name }}
+    {{ include "temporal.resourceLabels" (list . "admintools" "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/admintools-deployment.yaml:7]:
```suggestion
    {{- include "temporal.resourceLabels" (list . "admintools" "") | nindent 4 }}
```

File: charts/temporal/templates/admintools-deployment.yaml [hunk 5]
```diff
@@ -21,16 +15,7 @@ spec:
...
-        helm.sh/chart: {{ include "temporal.chart" . }}
-        app.kubernetes.io/managed-by: {{ .Release.Service }}
-        app.kubernetes.io/instance: {{ .Release.Name }}
-        app.kubernetes.io/version: {{ .Chart.AppVersion | replace "+" "_" }}
-        app.kubernetes.io/component: admintools
-        app.kubernetes.io/part-of: {{ .Chart.Name }}
-        {{- with $.Values.admintools.podLabels }}
-        {{- toYaml . | nindent 8 }}
-        {{- end }}
+        {{ include "temporal.resourceLabels" (list . "admintools" "pod") | nindent 8 }}
```
Comment (robholland) [inline charts/temporal/templates/admintools-deployment.yaml:18]:
```suggestion
        {{- include "temporal.resourceLabels" (list . "admintools" "pod") | nindent 8 }}
```

File: charts/temporal/templates/admintools-service.yaml [hunk 6]
```diff
@@ -4,13 +4,7 @@ kind: Service
...
   name: {{ include "temporal.componentname" (list . "admintools") }}
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" . }}
-    helm.sh/chart: {{ include "temporal.chart" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/version: {{ .Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/component: admintools
-    app.kubernetes.io/part-of: {{ .Chart.Name }}
+    {{ include "temporal.resourceLabels" (list . "admintools" "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/admintools-service.yaml:7]:
```suggestion
    {{- include "temporal.resourceLabels" (list . "admintools" "") | nindent 4 }}
```

File: charts/temporal/templates/frontend-ingress.yaml [hunk 7]
```diff
@@ -10,13 +10,7 @@ kind: Ingress
...
   name: {{ include "temporal.componentname" (list . "frontend") }}
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" . }}
-    helm.sh/chart: {{ include "temporal.chart" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/version: {{ .Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/component: frontend
-    app.kubernetes.io/part-of: {{ .Chart.Name }}
+    {{ include "temporal.resourceLabels" (list . "frontend" "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/frontend-ingress.yaml:13]:
```suggestion
    {{- include "temporal.resourceLabels" (list . "frontend" "") | nindent 4 }}
```

File: charts/temporal/templates/server-configmap.yaml [hunk 8]
```diff
@@ -6,12 +6,7 @@ kind: ConfigMap
 metadata:
   name: "{{ include "temporal.fullname" $ }}-config"
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" $ }}
-    helm.sh/chart: {{ include "temporal.chart" $ }}
-    app.kubernetes.io/managed-by: {{ $.Release.Service }}
-    app.kubernetes.io/instance: {{ $.Release.Name }}
-    app.kubernetes.io/version: {{ $.Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/part-of: {{ $.Chart.Name }}
+    {{ include "temporal.resourceLabels" (list . "" "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/server-configmap.yaml:9]:
```suggestion
    {{- include "temporal.resourceLabels" (list . "" "") | nindent 4 }}
```

File: charts/temporal/templates/server-deployment.yaml [hunk 9]
```diff
@@ -6,13 +6,7 @@ kind: Deployment
...
   name: {{ include "temporal.componentname" (list $ $service) }}
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" $ }}
-    helm.sh/chart: {{ include "temporal.chart" $ }}
-    app.kubernetes.io/managed-by: {{ $.Release.Service }}
-    app.kubernetes.io/instance: {{ $.Release.Name }}
-    app.kubernetes.io/version: {{ $.Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/component: {{ $service }}
-    app.kubernetes.io/part-of: {{ $.Chart.Name }}
+    {{ include "temporal.resourceLabels" (list $ $service "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/server-deployment.yaml:9]:
```suggestion
    {{- include "temporal.resourceLabels" (list $ $service "") | nindent 4 }}
```

File: charts/temporal/templates/server-deployment.yaml [hunk 10]
```diff
@@ -23,16 +17,7 @@ spec:
...
-        helm.sh/chart: {{ include "temporal.chart" $ }}
-        app.kubernetes.io/managed-by: {{ $.Release.Service }}
-        app.kubernetes.io/instance: {{ $.Release.Name }}
-        app.kubernetes.io/version: {{ $.Chart.AppVersion | replace "+" "_" }}
-        app.kubernetes.io/component: {{ $service }}
-        app.kubernetes.io/part-of: {{ $.Chart.Name }}
-        {{- with (default $.Values.server.podLabels $serviceValues.podLabels) }}
-        {{- toYaml . | nindent 8 }}
-        {{- end }}
+        {{ include "temporal.resourceLabels" (list $ $service "pod") | nindent 8 }}
```
Comment (robholland) [inline charts/temporal/templates/server-deployment.yaml:20]:
```suggestion
        {{- include "temporal.resourceLabels" (list $ $service "pod") | nindent 8 }}
```

File: charts/temporal/templates/server-dynamicconfigmap.yaml [hunk 11]
```diff
@@ -4,12 +4,7 @@ kind: ConfigMap
 metadata:
   name: "{{ include "temporal.fullname" . }}-dynamic-config"
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" . }}
-    helm.sh/chart: {{ include "temporal.chart" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/version: {{ .Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/part-of: {{ .Chart.Name }}
+    {{ include "temporal.resourceLabels" (list . "" "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/server-dynamicconfigmap.yaml:7]:
```suggestion
    {{- include "temporal.resourceLabels" (list . "" "") | nindent 4 }}
```

File: charts/temporal/templates/server-job.yaml [hunk 12]
```diff
@@ -4,26 +4,15 @@ kind: Job
...
   name: {{ include "temporal.componentname" (list . "schema") }}
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" . }}
-    helm.sh/chart: {{ include "temporal.chart" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/version: {{ .Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/component: database
-    app.kubernetes.io/part-of: {{ .Chart.Name }}
+    {{ include "temporal.resourceLabels" (list . "database" "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/server-job.yaml:7]:
```suggestion
    {{- include "temporal.resourceLabels" (list . "database" "") | nindent 4 }}
```

File: charts/temporal/templates/server-job.yaml [hunk 13]
```diff
@@ -4,26 +4,15 @@ kind: Job
...
       labels:
-        app.kubernetes.io/name: {{ include "temporal.name" . }}
-        helm.sh/chart: {{ include "temporal.chart" . }}
-        app.kubernetes.io/managed-by: {{ .Release.Service }}
-        app.kubernetes.io/instance: {{ .Release.Name }}
-        app.kubernetes.io/version: {{ .Chart.AppVersion | replace "+" "_" }}
-        app.kubernetes.io/component: database
-        app.kubernetes.io/part-of: {{ .Chart.Name }}
+        # Currently there are no labels for job type
+        {{ include "temporal.resourceLabels" (list . "database" "") | nindent 8 }}
```
Comment (robholland) [inline charts/temporal/templates/server-job.yaml:15]:
```suggestion
        {{- include "temporal.resourceLabels" (list . "database" "") | nindent 8 }}
```

File: charts/temporal/templates/server-pdb.yaml [hunk 14]
```diff
@@ -7,13 +7,7 @@ kind: PodDisruptionBudget
...
   name: {{ include "temporal.componentname" (list $ $service) }}-pdb
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" $ }}
-    helm.sh/chart: {{ include "temporal.chart" $ }}
-    app.kubernetes.io/managed-by: {{ $.Release.Service }}
-    app.kubernetes.io/instance: {{ $.Release.Name }}
-    app.kubernetes.io/version: {{ $.Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/component: {{ $service }}
-    app.kubernetes.io/part-of: {{ $.Chart.Name }}
+    {{ include "temporal.resourceLabels" (list . $service "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/server-pdb.yaml:10]:
```suggestion
    {{- include "temporal.resourceLabels" (list . $service "") | nindent 4 }}
```

File: charts/temporal/templates/server-secret.yaml [hunk 15]
```diff
@@ -13,15 +13,7 @@ kind: Secret
...
-    app.kubernetes.io/name: {{ include "temporal.name" $ }}
-    helm.sh/chart: {{ include "temporal.chart" $ }}
-    app.kubernetes.io/managed-by: {{ $.Release.Service }}
-    app.kubernetes.io/instance: {{ $.Release.Name }}
-    app.kubernetes.io/version: {{ $.Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/part-of: {{ $.Chart.Name }}
-  {{- with $.Values.server.secretLabels }}
-    {{- toYaml . | nindent 4 }}
-  {{- end }}
+    {{ include "temporal.resourceLabels" (list $ "" "secret") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/server-secret.yaml:16]:
```suggestion
    {{- include "temporal.resourceLabels" (list $ "" "secret") | nindent 4 }}
```

File: charts/temporal/templates/server-service-monitor.yaml [hunk 16]
```diff
@@ -7,13 +7,7 @@ kind: ServiceMonitor
...
   name: {{ include "temporal.componentname" (list $ $service) }}
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" $ }}
-    helm.sh/chart: {{ include "temporal.chart" $ }}
-    app.kubernetes.io/managed-by: {{ $.Release.Service }}
-    app.kubernetes.io/instance: {{ $.Release.Name }}
-    app.kubernetes.io/version: {{ $.Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/component: {{ $service }}
-    app.kubernetes.io/part-of: {{ $.Chart.Name }}
+    {{ include "temporal.resourceLabels" (list $ $service "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/server-service-monitor.yaml:10]:
```suggestion
    {{- include "temporal.resourceLabels" (list $ $service "") | nindent 4 }}
```

File: charts/temporal/templates/server-service.yaml [hunk 17]
```diff
@@ -4,13 +4,7 @@ kind: Service
...
   name: {{ include "temporal.componentname" (list . "frontend") }}
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" . }}
-    helm.sh/chart: {{ include "temporal.chart" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/version: {{ .Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/component: frontend
-    app.kubernetes.io/part-of: {{ .Chart.Name }}
+    {{ include "temporal.resourceLabels" (list $ "frontend" "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/server-service.yaml:7]:
```suggestion
    {{- include "temporal.resourceLabels" (list $ "frontend" "") | nindent 4 }}
```

File: charts/temporal/templates/server-service.yaml [hunk 18]
```diff
@@ -42,13 +36,7 @@ kind: Service
...
   name: {{ include "temporal.componentname" (list $ (printf "%s-headless" $service)) }}
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" $ }}
-    helm.sh/chart: {{ include "temporal.chart" $ }}
-    app.kubernetes.io/managed-by: {{ $.Release.Service }}
-    app.kubernetes.io/instance: {{ $.Release.Name }}
-    app.kubernetes.io/version: {{ $.Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/component: {{ $service }}
-    app.kubernetes.io/part-of: {{ $.Chart.Name }}
+    {{ include "temporal.resourceLabels" (list $ $service "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/server-service.yaml:39]:
```suggestion
    {{- include "temporal.resourceLabels" (list $ $service "") | nindent 4 }}
```

File: charts/temporal/templates/serviceaccount.yaml [hunk 19]
```diff
@@ -4,12 +4,7 @@ kind: ServiceAccount
 metadata:
   name: {{ include "temporal.serviceAccountName" . }}
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" . }}
-    helm.sh/chart: {{ include "temporal.chart" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/version: {{ .Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/part-of: {{ .Chart.Name }}
+    {{ include "temporal.resourceLabels" (list . "" "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/serviceaccount.yaml:7]:
```suggestion
    {{- include "temporal.resourceLabels" (list . "" "") | nindent 4 }}
```

File: charts/temporal/templates/tests/test.yaml [hunk 20]
```diff
@@ -3,13 +3,7 @@ kind: Pod
...
   name: "{{ include "temporal.fullname" . }}-test-wfl"
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" . }}
-    helm.sh/chart: {{ include "temporal.chart" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/version: {{ .Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/component: test
-    app.kubernetes.io/part-of: {{ .Chart.Name }}
+    {{ include "temporal.resourceLabels" (list $ "test" "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/tests/test.yaml:6]:
```suggestion
    {{- include "temporal.resourceLabels" (list $ "test" "") | nindent 4 }}
```

File: charts/temporal/templates/web-deployment.yaml [hunk 21]
```diff
@@ -4,13 +4,7 @@ kind: Deployment
...
   name: {{ include "temporal.componentname" (list . "web") }}
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" . }}
-    helm.sh/chart: {{ include "temporal.chart" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/version: {{ .Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/component: web
-    app.kubernetes.io/part-of: {{ .Chart.Name }}
+    {{ include "temporal.resourceLabels" (list . "web" "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/web-deployment.yaml:7]:
```suggestion
    {{- include "temporal.resourceLabels" (list . "web" "") | nindent 4 }}
```

File: charts/temporal/templates/web-deployment.yaml [hunk 22]
```diff
@@ -21,16 +15,7 @@ spec:
...
-        helm.sh/chart: {{ include "temporal.chart" . }}
-        app.kubernetes.io/managed-by: {{ .Release.Service }}
-        app.kubernetes.io/instance: {{ .Release.Name }}
-        app.kubernetes.io/version: {{ .Chart.AppVersion | replace "+" "_" }}
-        app.kubernetes.io/component: web
-        app.kubernetes.io/part-of: {{ .Chart.Name }}
-        {{- with .Values.web.podLabels }}
-        {{- toYaml . | nindent 8 }}
-        {{- end }}
+        {{ include "temporal.resourceLabels" (list $ "web" "pod") | nindent 8 }}
```
Comment (robholland) [inline charts/temporal/templates/web-deployment.yaml:18]:
```suggestion
        {{- include "temporal.resourceLabels" (list $ "web" "pod") | nindent 8 }}
```

File: charts/temporal/templates/web-ingress.yaml [hunk 23]
```diff
@@ -10,13 +10,7 @@ kind: Ingress
...
   name: {{ include "temporal.componentname" (list . "web") }}
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" . }}
-    helm.sh/chart: {{ include "temporal.chart" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/version: {{ .Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/component: web
-    app.kubernetes.io/part-of: {{ .Chart.Name }}
+    {{ include "temporal.resourceLabels" (list . "web" "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/web-ingress.yaml:13]:
```suggestion
    {{- include "temporal.resourceLabels" (list . "web" "") | nindent 4 }}
```

File: charts/temporal/templates/web-pdb.yaml [hunk 24]
```diff
@@ -5,13 +5,7 @@ kind: PodDisruptionBudget
...
   name: {{ include "temporal.componentname" (list . "web") }}-pdb
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" $ }}
-    helm.sh/chart: {{ include "temporal.chart" $ }}
-    app.kubernetes.io/managed-by: {{ $.Release.Service }}
-    app.kubernetes.io/instance: {{ $.Release.Name }}
-    app.kubernetes.io/version: {{ $.Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/component: web
-    app.kubernetes.io/part-of: {{ $.Chart.Name }}
+    {{ include "temporal.resourceLabels" (list . "web" "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/web-pdb.yaml:8]:
```suggestion
    {{- include "temporal.resourceLabels" (list . "web" "") | nindent 4 }}
```

File: charts/temporal/templates/web-service.yaml [hunk 25]
```diff
@@ -4,13 +4,7 @@ kind: Service
...
   name: {{ include "temporal.componentname" (list . "web") }}
   labels:
-    app.kubernetes.io/name: {{ include "temporal.name" . }}
-    helm.sh/chart: {{ include "temporal.chart" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/version: {{ .Chart.AppVersion | replace "+" "_" }}
-    app.kubernetes.io/component: web
-    app.kubernetes.io/part-of: {{ .Chart.Name }}
+    {{ include "temporal.resourceLabels" (list . "web" "") | nindent 4 }}
```
Comment (robholland) [inline charts/temporal/templates/web-service.yaml:7]:
```suggestion
    {{- include "temporal.resourceLabels" (list . "web" "") | nindent 4 }}
```

**Review by robholland — CHANGES_REQUESTED** [review]

**Comment by robholland** [comment]
Looking great, just a typo and some whitespace tweaks.

File: charts/temporal/templates/_helpers.tpl [hunk 26]
```diff
@@ -58,6 +58,61 @@ and we want to make sure that the component is included in the name.
...
+{{- end -}}
+{{- end -}}
+{{ include "temporal.additionalResourceLabels" $global }}
+{{- end -}}
+
+{{/*
+Additonal user specified labels for all resources
+*/}}
+{{- define "temporal.additionalResourceLabels" -}}
+{{- range $label_name, $label_value := .Values.addtionalLabels }}
```
[the remark this answers is not in the document]
Comment (sahilvv) [reply charts/temporal/templates/_helpers.tpl]: thank you for catching this 🙏

**Review by robholland — APPROVED** [review]

**Pull request merged**